scvi/external/cell2location/_module.py

```diff
@@ -0,0 +1,336 @@
+from torch.distributions.constraint_registry import biject_to, transform_to
+
+class SoftplusTransform(Transform):
+    r"""
+    Transform via the mapping :math:`\text{Softplus}(x) = \log(1 + \exp(x))`.
+    """
+    domain = constraints.real_vector
+    codomain = constraints.positive
+    
+    def __eq__(self, other):
+        return isinstance(other, SoftplusTransform)
+
+    def _call(self, x):
+        return (1+x.exp()).log()
+
+    def _inverse(self, y):
+        return y.expm1().log()
+      
+    def log_abs_det_jacobian(self, x, y):
+        return -(1+(-x).exp()).log()
+
+# replace exp transform with softplus https://github.com/pyro-ppl/numpyro/issues/855
+class _Positive(constraints.Constraint):
+    
+    def __init__(self):
+        
+        self.lower_bound = 0
+        super().__init__()
+    
+    def check(self, value):
+        return self.lower_bound < value
+
+    def __repr__(self):
+        fmt_string = self.__class__.__name__[1:]
+        fmt_string += '(lower_bound={})'.format(self.lower_bound)
+        return fmt_string
+
+@biject_to.register(_Positive)
+@transform_to.register(_Positive)
+def _transform_to_positive(constraint):
+    return SoftplusTransform()
+
+class ExpToSoftplus(pyro.poutine.messenger.Messenger):
+    
+    def __init__(self):
+        
+        super().__init__()
+    
+    def _process_message(self, msg):
+        if msg["type"] == "param" and msg["name"].endswith("_scale"):
+            msg["kwargs"]["constraint"] = _Positive()
+            
+_handler_name, ExpToSoftplusHandler = pyro.poutine.handlers._make_handler(ExpToSoftplus)
+
+# Define helper gamma distribution
+def Gamma(mu=None, sigma=None, alpha=None, beta=None, shape=None):
+    r"""
+    Function that converts mu/sigma Gamma distribution parametrisation into alpha/beta and
+    returns pyro Gamma distribution with an event of a given shape.
+    A thin wrapper over pyro.dist.Gamma.
+
+    :param mu: mean of Gamma distribution
+    :param sigma: variance of Gamma distribution. Note!!! this is :math:`\sigma^2` not :math:`\sigma`
+    :param alpha: shape parameter of Gamma distribution (mu ** 2 / sigma)
+    :param beta: rate parameter of Gamma distribution (mu / sigma)
+    :param shape: shape of the event / resulting variable. When None the shape is guessed based on input parameters
+
+    :return: pyro Gamma distribution class
+    """
+    if alpha is not None and beta is not None:
+        pass
+    elif mu is not None and sigma is not None:
+        alpha = mu ** 2 / sigma
+        beta = mu / sigma
+    else:
+        raise ValueError('Define (mu and sigma_sq) or (alpha and beta).')
+
+    if shape is None:
+        alpha = torch.tensor(alpha)
+        beta = torch.tensor(beta)
+    else:
+        alpha = torch.ones(shape) * torch.tensor(alpha)
+        beta = torch.ones(shape) * torch.tensor(beta)
+    return dist.Gamma(alpha, beta)
+
+class LocationModelLinearDependentWMultiExperiment(PyroBaseModuleClass):
```

Comment:
nitpicky but that's a very long class name

Reply:
How do you name different versions of the model?
with abbreviations? E.g. LMLDWME for this?